backend/app2/core/middleware/error_handler.py

```
@@ -0,0 +1,217 @@
+                # Attempt to extract response body for server errors
+                try:
+                    # Create a copy of the response to read the body without consuming it
+                    response_body = {}
+                    
+                    # Try to get error details from the response if possible
+                    if hasattr(response, "body"):
+                        body_bytes = await run_in_threadpool(lambda: response.body)
+                        if body_bytes:
+                            try:
+                                response_body = json.loads(body_bytes)
+                            except:
```

Ruff (0.8.2)

Do not use bare `except`

(E722)